core/org.eclipse.birt.core/src/org/eclipse/birt/core/util/EclipseUtil.java

```diff
@@ -0,0 +1,27 @@
+package org.eclipse.birt.core.util;
+
+import org.eclipse.core.runtime.Platform;
+import org.osgi.framework.Bundle;
+
+/**
+ * Eclipse and OSGi utilities
+ *
+ * @since 4.9
+ *
+ */
+public class EclipseUtil {
+
+	/**
+	 * Safely gets a bundle from Eclipse
+	 *
+	 * @param bundleId
+	 * @return the required bundle or null if the bundle could not be found or the
+	 *         platform is not running.
+	 */
+	public static Bundle getBundle(String bundleId) {
```

Comment:
Optional<Bundle> to avoid `null`
Then we can have clear understanding what to do on call site. For some cases we already have branches regarding exisitng/absent bundle, but not for all.

Reply:
This means changing the code everywhere instead of creating a simple redirect (which is a refactor). I don't have time for that.

Reply:
true, and it should not hold this fix from being applied